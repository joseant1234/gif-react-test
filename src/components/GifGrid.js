import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]) ;

    const { data:images, loading } = useFetchGifs(category);



    // con el arreglo vació solo se ejecuta una vez lo que está dentro del effect. Se ejecuta cuando se renderiza por primera vez el componente
    // useEffect(() => {
    //     getGifs(category).then(setImages(imgs));
    // }, [])

    // si se envía en el arreglo de dependecias, cada vez que se actualice category se va a ejecutar el effect
    // useEffect(() => {
    //     getGifs(category).then(setImages);
    // }, [category])

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>

            { loading && <p className="animate__animated animate__flash">Loading</p> }

            <div className="card-grid">

                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
