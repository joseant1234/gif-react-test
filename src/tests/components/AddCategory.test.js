import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from '../../components/AddCategory';

describe('Test to <AddCategory>', () => {

    const setCategories = () => {};

    test('should show correctly', () => {

        const wrapper = shallow(<AddCategory setCategories={setCategories} />);

        expect(wrapper).toMatchSnapshot(wrapper);
    })
})
