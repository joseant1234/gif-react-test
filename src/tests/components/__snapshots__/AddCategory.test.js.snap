// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <AddCategory> should show correctly 1`] = `Object {}`;
